File: sdk/r/R/support.R
#' Support: tickets, departments, news, knowledgebase
#' @param http HttpClient object
#' @return Resource object
#' @export
SupportResource <- function(http) {
  env <- new.env(parent = emptyenv())
  env$http <- http
  class(env) <- "SupportResource"
  env
}

#' GET /api/tickets - List support tickets
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
listTickets <- function(self, ...) {
  self$http$get("/api/tickets", list(...))
}

#' POST /api/tickets - Create new support ticket
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
createTicket <- function(self, ...) {
  self$http$post("/api/tickets", list(...))
}

#' GET /api/tickets/{number} - Get ticket details with replies
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
getTicket <- function(self, ...) {
  self$http$get("/api/tickets/{number}", list(...))
}

#' POST /api/tickets/{number} - Reply to ticket
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
reply <- function(self, ...) {
  self$http$post("/api/tickets/{number}", list(...))
}

#' GET /api/ticket/attachment/{file} - Get ticket attachment
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
getAttachment <- function(self, ...) {
  self$http$get("/api/ticket/attachment/{file}", list(...))
}

#' PUT /api/tickets/{number}/open - Re-open closed ticket
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
reopen <- function(self, ...) {
  self$http$put("/api/tickets/{number}/open", list(...))
}

#' PUT /api/tickets/{number}/close - Close ticket
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
close <- function(self, ...) {
  self$http$put("/api/tickets/{number}/close", list(...))
}

#' GET /api/ticket/departments - List ticket departments
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
listDepartments <- function(self, ...) {
  self$http$get("/api/ticket/departments", list(...))
}

#' GET /api/news - List news
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
listNews <- function(self, ...) {
  self$http$get("/api/news", list(...))
}

#' GET /api/news/{news_id} - Get news item details
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
getNews <- function(self, ...) {
  self$http$get("/api/news/{news_id}", list(...))
}

#' GET /api/knowledgebase - List knowledgebase categories
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
listKnowledgebaseCategories <- function(self, ...) {
  self$http$get("/api/knowledgebase", list(...))
}

#' GET /api/knowledgebase/{category_id} - Get category with articles
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
getKnowledgebaseCategory <- function(self, ...) {
  self$http$get("/api/knowledgebase/{category_id}", list(...))
}

#' GET /api/knowledgebase/article/{article_id} - Get article details
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
getKnowledgebaseArticle <- function(self, ...) {
  self$http$get("/api/knowledgebase/article/{article_id}", list(...))
}

File: sdk/c/src/support.c
#include "hitechcloud/support.h"
#include <stdlib.h>


/* GET /api/tickets - List support tickets */
char *hc_support_list_tickets(hc_http_client_t *client) {
    return hc_get(client, "/api/tickets", NULL);
}

/* POST /api/tickets - Create new support ticket */
char *hc_support_create_ticket(hc_http_client_t *client) {
    return hc_post(client, "/api/tickets", NULL);
}

/* GET /api/tickets/{number} - Get ticket details with replies */
char *hc_support_get_ticket(hc_http_client_t *client) {
    return hc_get(client, "/api/tickets/{number}", NULL);
}

/* POST /api/tickets/{number} - Reply to ticket */
char *hc_support_reply(hc_http_client_t *client) {
    return hc_post(client, "/api/tickets/{number}", NULL);
}

/* GET /api/ticket/attachment/{file} - Get ticket attachment */
char *hc_support_get_attachment(hc_http_client_t *client) {
    return hc_get(client, "/api/ticket/attachment/{file}", NULL);
}

/* PUT /api/tickets/{number}/open - Re-open closed ticket */
char *hc_support_reopen(hc_http_client_t *client) {
    return hc_put(client, "/api/tickets/{number}/open", NULL);
}

/* PUT /api/tickets/{number}/close - Close ticket */
char *hc_support_close(hc_http_client_t *client) {
    return hc_put(client, "/api/tickets/{number}/close", NULL);
}

/* GET /api/ticket/departments - List ticket departments */
char *hc_support_list_departments(hc_http_client_t *client) {
    return hc_get(client, "/api/ticket/departments", NULL);
}

/* GET /api/news - List news */
char *hc_support_list_news(hc_http_client_t *client) {
    return hc_get(client, "/api/news", NULL);
}

/* GET /api/news/{news_id} - Get news item details */
char *hc_support_get_news(hc_http_client_t *client) {
    return hc_get(client, "/api/news/{news_id}", NULL);
}

/* GET /api/knowledgebase - List knowledgebase categories */
char *hc_support_list_knowledgebase_categories(hc_http_client_t *client) {
    return hc_get(client, "/api/knowledgebase", NULL);
}

/* GET /api/knowledgebase/{category_id} - Get category with articles */
char *hc_support_get_knowledgebase_category(hc_http_client_t *client) {
    return hc_get(client, "/api/knowledgebase/{category_id}", NULL);
}

/* GET /api/knowledgebase/article/{article_id} - Get article details */
char *hc_support_get_knowledgebase_article(hc_http_client_t *client) {
    return hc_get(client, "/api/knowledgebase/article/{article_id}", NULL);
}

